src/libclipper/include/clipper/rpc_service.hpp

```diff
@@ -31,6 +32,14 @@ using RPCResponse = std::pair<const int, vector<uint8_t>>;
 using RPCRequest =
 std::tuple<const int, const int, const std::vector<std::vector<uint8_t>>, const long>;
 
+template <class T>
+class VectorHasher {
+ public:
+  size_t operator()(const vector<T> &v) const {
+    return boost::hash_range(v.begin(), v.end());
+  }
+};
+
```

Comment:
I moved the method `<T> primitive_input_hash(std::vector<T)` in `datatypes.cpp` to a generic `hash_vector` method in `util.hpp`. The new method makes use of `boost::hash_range()` and is now used for hashing input vectors for serialization as well as for the rpc use case of hashing connection ids.